src/coreclr/inc/corhlprpriv.h

```diff
@@ -225,71 +225,45 @@ class CQuickMemoryBase
         iSize = cbTotal;
     }
 
-
-    // Convert UTF8 string to UNICODE string, optimized for speed
-    HRESULT ConvertUtf8_UnicodeNoThrow(const char * utf8str)
-    {
-        bool allAscii;
-        DWORD length;
-
-        HRESULT hr = FString::Utf8_Unicode_Length(utf8str, & allAscii, & length);
-
-        if (SUCCEEDED(hr))
-        {
-            LPWSTR buffer = (LPWSTR) AllocNoThrow((length + 1) * sizeof(WCHAR));
-
-            if (buffer == NULL)
-            {
-                hr = E_OUTOFMEMORY;
-            }
-            else
-            {
-                hr = FString::Utf8_Unicode(utf8str, allAscii, buffer, length);
-            }
-        }
-
-        return hr;
-    }
-
     // Convert UTF8 string to UNICODE string, optimized for speed
     void ConvertUtf8_Unicode(const char * utf8str)
     {
-        bool allAscii;
-        DWORD length;
-
-        HRESULT hr = FString::Utf8_Unicode_Length(utf8str, & allAscii, & length);
-
-        if (SUCCEEDED(hr))
+        size_t sourceLen = 0;
+        size_t destLen = 0;
+        if (utf8str != NULL)
         {
-            LPWSTR buffer = (LPWSTR) AllocThrows((length + 1) * sizeof(WCHAR));
-
-            hr = FString::Utf8_Unicode(utf8str, allAscii, buffer, length);
+            sourceLen = strlen(utf8str) + 1;
+            destLen = minipal_get_length_utf8_to_utf16(utf8str, sourceLen, /* flags */ 0);
         }
 
-        if (FAILED(hr))
+        CHAR16_T* buffer = (CHAR16_T*) AllocThrows((destLen + 1) * sizeof(CHAR16_T));
+        buffer[destLen] = W('\0');
+
+        if (destLen > 0
+            && !minipal_convert_utf8_to_utf16(utf8str, sourceLen, buffer, destLen, /* flags */ 0))
         {
-            ThrowHR(hr);
+            ThrowHR(EMAKEHR(GetLastError()));
```

Comment:
`minipal_convert_utf8_to_utf16` sets `errno`. I do not think it is correct to use `GetLastError` here. `GetLastError` is not same as `errno` on Windows.

Reply:
Shoot, your're right. I was following the PAL implementation model. This needs to use `errno`. Opinions on instead of this approach implement it in terms of the Win32 APIs and rely on the PAL itself?

Reply:
I think we want to be eliminating dependencies on Win32 PAL in cross-platform code.

Reply:
Okay. Then will update in terms of `errno`.